Sources/Rendering/Core/CellPicker/index.js

```diff
@@ -317,17 +335,20 @@ function vtkCellPicker(publicAPI, model) {
     const x1 = [0, 0, 0];
     const x2 = [0, 0, 0];
     for (let i = 0; i < 3; i++) {
-      x1[i] = (p1[i] - origin[i]) / spacing[i];
-      x2[i] = (p2[i] - origin[i]) / spacing[i];
+      x1[i] = (q1[i] - origin[i]) / spacing[i];
+      x2[i] = (q2[i] - origin[i]) / spacing[i];
     }
     const x = [0, 0, 0, 0];
     const xi = [0, 0, 0];
 
     const sliceSize = dims[1] * dims[0];
     const rowSize = dims[0];
-    const nSteps = 100;
+    // here the step is the 1 over the distance between volume index location x1 and x2
+    const step = 1 / Math.sqrt(vtkMath.distance2BetweenPoints(x1, x2));
     let insideVolume;
-    for (let t = t1; t < t2; t += 1 / nSteps) {
+    // here we reinterpret the t value as the distance between x1 and x2
+    // When calculating the tMin, we weight t between t1 and t2 values
+    for (let t = 0; t < 1; t += step) {
       // calculate the location of the point
       insideVolume = true;
```

Comment:
it seems that `insideVolume` will always be equal to `true`. How useful is this variable ?

Reply:
If out of bounds in one dimension it could be false and the code didnt consider the particular  volume index coordinate

Reply:
my bad, I didn't read the code afterwards